### PR TITLE
[Regression Fix] Readd timeline caching

### PR DESCRIPTION
**Pull request by Mikescops**
Fixing a regression following #468 after update of the vue-virtual-grid package.

The caching of the timeline is now used again. It helped simplifying some part of the logic too.

### CODE REVIEW

File: src/components/Loader.vue [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+ -
+ - You should have received a copy of the GNU Affero General Public License
+ - along with this program. If not, see <http://www.gnu.org/licenses/>.
+ -
+ -->
+
+<template>
+	<div class="loader" icon="loading">
+		<slot name="icon">
+			<span :class="['loading']" />
```
Comment (skjnldsv) [inline src/components/Loader.vue:26]:
```suggestion
			<span class="icon-loading" />
```

File: src/components/Loader.vue [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+ - MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ - GNU Affero General Public License for more details.
+ -
+ - You should have received a copy of the GNU Affero General Public License
+ - along with this program. If not, see <http://www.gnu.org/licenses/>.
+ -
+ -->
+
+<template>
+	<div class="loader" icon="loading">
```
Comment (skjnldsv) [inline src/components/Loader.vue:24]: `icon="loading"` this doesn't do anything, right?

Comment (Mikescops) [reply src/components/Loader.vue:24]: oups, that's a test leftover

**Comment by Mikescops** [comment]
@skjnldsv done

File: src/views/Timeline.vue [hunk 3]
```diff
@@ -144,65 +222,25 @@ export default {
 			try {
 				// Load next batch of images
 				const files = await request(this.onlyFavorites, {
-					page: params.offset, // offset is incremented +1 by the virtualgrid lib
+					page: this.page, // offset is incremented +1 by the virtualgrid lib
```
Comment (skjnldsv) [inline src/views/Timeline.vue:225]:
```suggestion
					page: this.page,
```

**Review by skjnldsv — APPROVED** [review]
Tiny change, all good code-wise :)
One question though, why use Promise.resolve ?
Returning should be enough in those cases,

**Comment by Mikescops** [comment]
The lib expect a Promise so that it can do other actions in the meantime.

**Comment by skjnldsv** [comment]
> The lib expect a Promise so that it can do other actions in the meantime.

The async function resolves themselves with return or throw.
Async functions are already promises :)

**Comment by skjnldsv** [comment]
This returns a promise:

```
function resolveAfter2Seconds() {
  return new Promise(resolve => {
    setTimeout(() => {
      resolve('resolved');
    }, 2000);
  });
}
```

`Promise.resolve` have no context here. I don't think that is necessary

**Comment by Mikescops** [comment]
hmmm I need to better look at it

**Pull request merged**